test/test_wand_data.cpp

```diff
@@ -0,0 +1,109 @@
+#define CATCH_CONFIG_MAIN
+#include "catch2/catch.hpp"
+
+#include <functional>
+
+#include "test_common.hpp"
+
+#include "pisa_config.hpp"
+#include "index_types.hpp"
+#include "query/queries.hpp"
+#include "wand_data_range.hpp"
+#include "scorer/score_function.hpp"
+
+using namespace pisa;
+
+TEST_CASE("wand_data_range") {
+    typedef wand_data_range<64, 1024, bm25> WandTypeRange;
+    typedef wand_data<bm25, WandTypeRange> WandType;
+    using Scorer = bm25;
+
+    binary_freq_collection collection(PISA_SOURCE_DIR "/test/test_data/test_collection");
+    binary_collection      document_sizes(PISA_SOURCE_DIR "/test/test_data/test_collection.sizes");
+    WandType          wdata_range(document_sizes.begin()->begin(),
+                              collection.num_docs(),
+                              collection);
+    SECTION( "Precomputed block-max scores" ) {
+        size_t i = 0;
+        for (auto const &seq: collection) {
```

Comment:
What you want to do here: include `range/v3/view/enumerate.hpp` and use:
```cpp
for (auto &&[term_id, seq] : ranges::view::enumerate(collection))
```

[range-v3](https://github.com/ericniebler/range-v3) is an implementation of ranges that will come with C++20 standard. Here's a nice presentation if you're interested to find out more https://www.youtube.com/watch?v=mFUXNMfaciE&t=533s